app/services/DateService.scala

```diff
@@ -90,6 +90,14 @@ class DateService @Inject()(implicit val frontendAppConfig: FrontendAppConfig) e
     TaxYear.forYearEnd(yearEnd)
   }
 
+  def isWithin30Days(date: Option[LocalDate]): Boolean = {
+    val currentDate = getCurrentDate
+    date match {
+      case Some(dueDate) => dueDate.minusDays(30).isBefore(currentDate)
+      case None => false
```

Comment:
I guess this would a strange outcome of this function if there was no actual date provided and we return false